benchmark/function_args.cc

```diff
@@ -89,65 +92,114 @@ static Napi::Object Init(Napi::Env env, Napi::Object exports) {
+  NAPI_THROW_IF_FAILED_VOID(env, status);
 
   status = napi_create_function(env,
                                 "oneArgFunction",
                                 NAPI_AUTO_LENGTH,
                                 OneArgFunction_Core,
                                 nullptr,
                                 &one_arg_function);
-  NAPI_THROW_IF_FAILED(env, status, Napi::Object());
+  NAPI_THROW_IF_FAILED_VOID(env, status);
 
   status = napi_create_function(env,
                                 "twoArgFunction",
                                 NAPI_AUTO_LENGTH,
                                 TwoArgFunction_Core,
                                 nullptr,
                                 &two_arg_function);
-  NAPI_THROW_IF_FAILED(env, status, Napi::Object());
+  NAPI_THROW_IF_FAILED_VOID(env, status);
 
   status = napi_create_function(env,
                                 "threeArgFunction",
                                 NAPI_AUTO_LENGTH,
                                 ThreeArgFunction_Core,
                                 nullptr,
                                 &three_arg_function);
-  NAPI_THROW_IF_FAILED(env, status, Napi::Object());
+  NAPI_THROW_IF_FAILED_VOID(env, status);
 
   status = napi_create_function(env,
                                 "fourArgFunction",
                                 NAPI_AUTO_LENGTH,
                                 FourArgFunction_Core,
                                 nullptr,
                                 &four_arg_function);
-  NAPI_THROW_IF_FAILED(env, status, Napi::Object());
-
-  Napi::Object core = Napi::Object::New(env);
-  core["noArgFunction"] = Napi::Value(env, no_arg_function);
-  core["oneArgFunction"] = Napi::Value(env, one_arg_function);
-  core["twoArgFunction"] = Napi::Value(env, two_arg_function);
-  core["threeArgFunction"] = Napi::Value(env, three_arg_function);
-  core["fourArgFunction"] = Napi::Value(env, four_arg_function);
-  exports["core"] = core;
-
-  Napi::Object cplusplus = Napi::Object::New(env);
-  cplusplus["noArgFunction"] = Napi::Function::New(env, NoArgFunction);
-  cplusplus["oneArgFunction"] = Napi::Function::New(env, OneArgFunction);
-  cplusplus["twoArgFunction"] = Napi::Function::New(env, TwoArgFunction);
-  cplusplus["threeArgFunction"] = Napi::Function::New(env, ThreeArgFunction);
-  cplusplus["fourArgFunction"] = Napi::Function::New(env, FourArgFunction);
-  exports["cplusplus"] = cplusplus;
-
-  Napi::Object templated = Napi::Object::New(env);
-  templated["noArgFunction"] = Napi::Function::New<NoArgFunction>(env);
-  templated["oneArgFunction"] = Napi::Function::New<OneArgFunction>(env);
-  templated["twoArgFunction"] = Napi::Function::New<TwoArgFunction>(env);
-  templated["threeArgFunction"] = Napi::Function::New<ThreeArgFunction>(env);
-  templated["fourArgFunction"] = Napi::Function::New<FourArgFunction>(env);
-  exports["templated"] = templated;
-
-  return exports;
-}
+  NAPI_THROW_IF_FAILED_VOID(env, status);
```

Comment:
These changes mean we won't be able to run the benchmarks if the target is less than NAPI_VERSION 6.

I wonder if the changes to the benchmark code should be additive so that we get new numbers if the target is NAPI_VERSION 6 or greater as opposed to replacing the existing ones.

That would also let us keep an eye on the delta between using the additional wrapper or not.